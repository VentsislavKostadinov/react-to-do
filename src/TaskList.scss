.container {
    text-align: center;
    margin-top: 3rem;

    .task-list {
        margin-top: 2rem;
        .list-group-item {
            border: none;
        }
    }
}
