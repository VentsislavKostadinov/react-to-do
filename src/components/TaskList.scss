@import '../style/variables';
.container {
    text-align: center;
    padding-top: 3rem;

    .col {
        p {
            padding-top: 2rem !important;
        }

        .add-input {
            border-radius: 5px 0 0 5px !important;
        }
        .form-check-input {
            &:hover {
                cursor: pointer;
            }
        }
        hr {
            margin: 0;
            color: $white-color;
        }

        .task-list {
            margin-top: 2rem;
            .list-group-item {
                border: none;
            }
        }
    }
}
