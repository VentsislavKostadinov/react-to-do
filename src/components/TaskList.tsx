import React, { useState } from 'react'
import { TaskListProps } from '../model/TaskListProps.types'
import { Container, Form, Row, Col } from 'react-bootstrap'
import { CommonButton } from '../common/CommonButton'
import { CommonHeadline } from '../common/CommonHeadline'
import { CommonInput } from '../common/CommonInput'
import { CommonInvalidValiadationFeedback } from '../common/CommonInvalidValiadationFeedback'
import { Task } from './Task'
import { TaskProps } from '../model/TaskProps.types'
import classes from '../style/TaskList.module.scss'

export const TaskList = ({
    tasks,
    onAdd,
    onEdit,
    onDelete,
    onComplete,
}: TaskListProps) => {
    const [newTask, setNewTask] = useState<string>('')
    const [validated, setValidated] = useState<boolean>(false)

    console.log(newTask.length)

    const handleSetNewTask = (e: React.ChangeEvent<HTMLInputElement>) => {
        setNewTask(e.target.value)
    }

    const handleSubmit = (e: React.FormEvent) => {
        e.preventDefault()

        if (newTask === '') {
            setValidated(true)
        } else {
            setValidated(false)
            onAdd(newTask)
            setNewTask('')
        }
    }

    return (
        <Container className={classes.container}>
            <Row>
                <Col className={classes.col}>
                    <Form
                        noValidate
                        validated={validated}
                        onSubmit={handleSubmit}
                    >
                        <CommonInput
                            type="text"
                            value={newTask}
                            handleChange={handleSetNewTask}
                            placeholder="Enter a description"
                            className={classes.addInput}
                            dataTestid="add-input"
                        >
                            <CommonButton
                                variant="info"
                                text="Add"
                                type="submit"
                            />
                            <CommonInvalidValiadationFeedback text="Please type a description" />
                        </CommonInput>
                    </Form>
                    <CommonHeadline title="Task list" heading="h2" />
                    <div>
                        {tasks.length !== 0 ? (
                            tasks.map(
                                (task: TaskProps | any, index: number) => (
                                    <div key={index}>
                                        <Task
                                            key={index}
                                            task={task}
                                            onEdit={onEdit}
                                            onDelete={onDelete}
                                            onComplete={onComplete}
                                        />
                                        <hr />
                                    </div>
                                ),
                            )
                        ) : (
                            <CommonHeadline
                                title="You list is empty"
                                heading="h3"
                            />
                        )}
                    </div>
                </Col>
            </Row>
        </Container>
    )
}
