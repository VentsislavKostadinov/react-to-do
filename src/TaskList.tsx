import React, { useState } from 'react'
import { TaskListProps } from './model/TaskListProps.types'
import { Container, Form } from 'react-bootstrap'
import { CommonButton } from './common/CommonButton'
import { CommonHeadline } from './common/CommonHeadline'
import { CommonInput } from './common/CommonInput'
import { CommonInvalidValiadationFeedback } from './common/CommonInvalidValiadationFeedback'
import { Task } from './Task'
import { TaskProps } from './model/TaskProps.types'
import './TaskList.scss'

export const TaskList = ({
    tasks,
    onAdd,
    onEdit,
    onDelete,
    onComplete,
}: TaskListProps) => {
    const [newTask, setNewTask] = useState<string>('')
    const [validated, setValidated] = useState<boolean>(false)

    const handleSetNewTask = (e: React.ChangeEvent<HTMLInputElement>) => {
        setNewTask(e.target.value)
    }

    const handleSubmit = (event: React.FormEvent) => {
        event.preventDefault()

        if (newTask === '') {
            setValidated(true)
        } else {
            setValidated(false)
            onAdd(newTask)
            setNewTask('')
        }
    }

    return (
        <div>
            <Container>
                <CommonHeadline title="To Do App" heading="h2" />
                <Form noValidate validated={validated} onSubmit={handleSubmit}>
                    <CommonInput
                        type="text"
                        value={newTask}
                        handleChange={handleSetNewTask}
                        placeholder="Enter a description"
                        className="add-input"
                        dataTestid="add-input"
                    >
                        <CommonButton
                            variant="primary"
                            text="Add"
                            type="submit"
                        />
                        <CommonInvalidValiadationFeedback text="Please type a description" />
                    </CommonInput>
                </Form>
                {tasks.length !== 0 ? (
                    tasks.map((task: TaskProps | any, index: number) => {
                        return (
                            <Task
                                key={index}
                                task={task}
                                onEdit={onEdit}
                                onDelete={onDelete}
                                onComplete={onComplete}
                            />
                        )
                    })
                ) : (
                    <CommonHeadline title="You list is empty" heading="h3" />
                )}
            </Container>
        </div>
    )
}
