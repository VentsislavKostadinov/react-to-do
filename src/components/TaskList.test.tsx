import React from 'react'
import { TaskList } from './TaskList'
import { TaskListProps } from '../model/TaskListProps.types'
import { render, screen, fireEvent } from '@testing-library/react'

describe('TaskList component', () => {
    const taskListProps: TaskListProps = {
        tasks: [
            {
                task: {
                    id: 4534534,
                    description: 'text description',
                    completed: false,
                },
                onEdit: jest.fn,
                onComplete: jest.fn,
                onDelete: jest.fn,
            },
        ],
        onAdd: jest.fn,
        onEdit: jest.fn,
        onDelete: jest.fn,
        onComplete: jest.fn,
    }

    it('render correctly', () => {
        const { asFragment } = render(<TaskList {...taskListProps} />)
        expect(asFragment).toMatchSnapshot()
    })

    it('displayed correctly', () => {
        render(<TaskList {...taskListProps} />)

        const container = document.querySelectorAll('.container')

         expect(container[1].innerHTML).toContain('Edit')
         expect(container[1].innerHTML).toContain('Delete')
    })
})
