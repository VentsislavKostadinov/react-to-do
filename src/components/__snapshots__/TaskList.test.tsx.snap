// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList component render correctly 1`] = `[Function]`;
